src/python/TaskWorker/Worker.py

```diff
@@ -36,13 +36,17 @@ def addTaskLogHandler(logger, username, taskname):
     #set the logger to save the tasklog
     formatter = logging.Formatter("%(asctime)s:%(levelname)s:%(module)s:%(message)s")
     taskdirname = "logs/tasks/%s/" % username
-    if not os.path.isdir(taskdirname):
+    try:
         os.mkdir(taskdirname)
+    except OSError as ose:
+        if ose.errno != 17: #ignore the "Directory already exists error" but print other errors traces
+            logger.exception("Cannot set task hanlder logfile for task %s. Ignoring and continuing normally." % taskname)
```

Comment:
The log/task/taskname.log file will not be created (and not present in the crabcache for the monitor). Apart from that the task will be processed as normal. That's the current behavior.

The alternative is to completely stop the subprocess and leave the task stuck in QUEUED. Which is basically what happened in production for those 4 tasks. Setting the task to FAILED is a bit tricky (it's not enough to raise TaskWorkerException as usual because we are not handling actions yet but setting things up).

Keep in mind this never happened in the past (or we would have noticed tasks stuyck in QUEUED early on), and the recent thing of the directory already existing is being catched. I don't see how else it can fail that is not a disaster-like case (disk full/unaccessible) that would be noticed in any way. So in general I'd say "ignoring"/"stopping the subprocess" does not make any difference :)
But let me know if you prefer to stop the subprocess and leave the task to QUEUED.